usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MedicationManagerClient.java

```diff
@@ -0,0 +1,216 @@
+                                                     DateTime initialDate,
+                                         DateTime finalDate) throws ExecutionException, InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response =
+                taskManager.execute(BASE_URL + owner + dash + petName + "/between/" + initialDate
+                        + dash + finalDate, accessToken).get();
+        List<Medication> MedicationList = new ArrayList<>();
+        if (response.length() > 2) {
+            String jsonArray = response.substring(1, response.length() - 1);
+            String[] Medications = jsonArray.split(",\\{");
+            MedicationList.add(GSON.fromJson(Medications[0], Medication.class));
+            for (int i = 1; i < Medications.length; i++) {
+                Medications[i] = "{" + Medications[i];
+                MedicationList.add(GSON.fromJson(Medications[i], Medication.class));
+            }
+        }
+        return MedicationList;
+    }
+
+    /**
+     * Updates the Medication's field.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param date Date of the medication
+     * @param name Name of the medication
+     * @param field Name of the field to update
+     * @param value Value the field will have
+     * @return The response code
+     * @throws ExecutionException When the update fails
+     * @throws InterruptedException When the update is interrupted
+     */
+    public int updateMedicationField(String accessToken, String owner, String petName, DateTime date,
+                                     String name, String field,  Object value)
+            throws ExecutionException, InterruptedException {
+        checkCorrectType(field, value);
+        Map<String, Object> reqData = new HashMap<>();
+        reqData.put("value", value);
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(PUT);
+        taskManager.setReqBody(new JSONObject(reqData));
+        StringBuilder response = taskManager.execute(BASE_URL + owner + dash + petName + dash
+                + date + dash + name + dash + field, accessToken).get();
+        return Integer.parseInt(response.toString());
+    }
+
+    /**
+     * Checks that the new value is of the correct type.
+     * @param field The field to update
+     * @param value The new value
+     * @throws IllegalArgumentException When an invalid field value is passed
+     */
+    private void checkCorrectType(String field, Object value) {
+        if (field.equals(QUANTITY) && !(value instanceof Double)) {
+            throw new IllegalArgumentException("New value must be a Double");
+        }
+        if (field.equals(DURATION) && !(value instanceof Integer)) {
+            throw new IllegalArgumentException("New value must be an Integer");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String "New value must be an Integer" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403182)